testindex/summary.py

```diff
@@ -21,39 +21,46 @@ def add_warning(self, msg):
 class Summary(object):
     """This class summarizes the tests"""
 
-    _summary = {}
-
-    def __init__(self):
+    def __init__(self, summary_dump):
+        self.global_errors = []
+        self._summary = {}
+        self._summary_dump = summary_dump
         pass
 
     @staticmethod
     def _get_entry_hash(file_name, entry):
 
         return hashlib.sha256(file_name + str(entry)).hexdigest()
 
-    @staticmethod
-    def pakage_private_get_summary_object(file_name, entry):
+    def get_summary_object(self, file_name, entry):
 
-        if file_name not in Summary._summary:
-            Summary._summary[file_name] = {}
+        if file_name not in self._summary:
+            self._summary[file_name] = {}
 
         entry_hash = Summary._get_entry_hash(file_name, entry)
 
-        if entry_hash not in Summary._summary[file_name]:
-            Summary._summary[file_name][entry_hash] = {
+        if entry_hash not in self._summary[file_name]:
+            self._summary[file_name][entry_hash] = {
                 "entry": str(entry),
                 "errors": [],
                 "warnings": []
             }
```

Comment:
I strive for readability infact i will be doing a lot more refactoring for making the code more readable in later prs. :)